api/pkg/client/mlflow/mlflow.go

```diff
@@ -22,12 +25,24 @@ type mlflowService struct {
 	Config          Config
 }
 
-func NewMlflowService(httpClient *http.Client, config Config, artifactService artifact.Service) Service {
+func NewMlflowService(httpClient *http.Client, config Config) (Service, error) {
+	var artifactService artifact.Service
+	if config.ArtifactServiceType == "nop" {
+		artifactService = artifact.NewNopArtifactService()
+	} else if config.ArtifactServiceType == "gcs" {
+		api, err := storage.NewClient(config.ArtifactServiceGcsConfig.Ctx)
```

Comment:
Hmm I'm thinking that since we'll almost always be creating a new context initialise the GCS client, there's probably no need pass a context to this function as part of the `config` argument. We can instead just create a new context `api, err := storage.NewClient(context.Background()` straight away. This also saves us the need to specific a `ArtifactServiceGcsConfig ` field for the `mlflow.Config` struct.